test/system/flows/user_registration_test.rb

```diff
@@ -49,52 +49,59 @@ class UserRegistrationTest < ApplicationSystemTestCase
 
     test "user registers via Github" do
       OmniAuth.config.test_mode = true
-      OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new({
-                                                                    provider: "github",
-                                                                    uid: "111",
-                                                                    info: {
-                                                                      email: "user@exercism.io",
-                                                                      name: "Name",
-                                                                      nickname: "user22"
-                                                                    }
-                                                                  })
+      OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new(
+        provider: "github",
+        uid: "111",
+        info: {
+          email: "user@exercism.io",
+          name: "Name",
+          nickname: "user22"
+        }
+      )
       visit new_user_registration_path
       click_on "Sign in with GitHub"
 
       refute_text "Please confirm your email"
       assert_text "Onboarding"
 
+    ensure
       OmniAuth.config.test_mode = false
     end
 
     test "user sees errors when registering via Github" do
       OmniAuth.config.test_mode = true
-      OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new({
-                                                                    provider: "github",
-                                                                    uid: "111",
-                                                                    info: {
-                                                                      email: nil,
-                                                                      name: "Name",
-                                                                      nickname: "user22"
-                                                                    }
-                                                                  })
+      OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new(
+        provider: "github",
+        uid: "111",
+        info: {
+          email: nil,
+          name: "Name",
+          nickname: "user22"
+        }
+      )
       visit new_user_registration_path
       click_on "Sign in with GitHub"
 
       assert_text "Sorry, we could not authenticate you from GitHub."
 
+    ensure
```

Comment:
I don't know if we actually **need** to turn test mode back on (I suspect we don't). But if we do, we should add the `ensure` in case the test exceptions.